optimum/pipelines/diffusers/watermark.py

```diff
@@ -18,6 +18,10 @@ def apply_watermark(self, images: np.array):
         if images.shape[-1] < 256:
             return images
 
+        # cv2 doesn't support float16
+        if images.dtype == np.float16:

```

Comment:
I am unsure whether the cast should be done in the watermarker or in the mixin lines linked in the description. 
Decided to do it in the watermarker since custom watermarkers are probably expected to deal with fp16 images themselves.

Reply:
The diffusers library also casts it to float here https://github.com/huggingface/diffusers/blob/main/src/diffusers/pipelines/stable_diffusion_xl/watermark.py#L29